src/app/core/press-coverage.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { throwError } from 'rxjs';
+import { catchError, retry } from 'rxjs/operators';
+
+export class Mention {
+  constructor(public title: string, public link: string, public icon: string) { }
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PressCoverageService {
+  private _mentions: Mention[];
+
+  get mentions() {
+    return this._mentions;
+  }
+  constructor(private http: HttpClient) { }
+
+  getMentions(): Promise<Mention[]> {
+    if (this._mentions) {
+      return Promise.resolve(this._mentions);
+    } else {
+      return this.http.get('assets/images/press/mentions.json', { responseType: 'json' })
+        .pipe(
+          retry(3),
+          catchError(this.handleError)
+        )
+        .toPromise()
+        .then((val: Mention[]) => {
+          this._mentions = [];
+          val.forEach(v => {
+            this._mentions.push(new Mention(v.title, v.link, v.icon));
```

Comment:
Couldn't we make mentions a interface and say that the http request gets a mentions array?  Also, couldn't we make getMentions() a async function and await the http call?

Reply:
I made the getMentions function async. Right now this is just retrieving from an internal json store so I think I would like to keep Mention a class with the constructor at the top of this service. If we choose to create a service in the future that has a service we are returning from we can transition to an interface.